application/tests/AuthHelpers.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Tests;
+
+use Firebase\JWT\JWT;
+
+trait AuthHelpers
+{
+    public static function generateAccessToken($tolkevaravPayload = [])
+    {
+        // TODO: would be good to have full example JWT here with
+        // TODO: all relevant claims to simulate real JWT.
+        // TODO: This JWT should be overwrittable to support
+        // TODO: different edge cases.
+        $payload = [
+            'tolkevarav' => collect([
+                'userId' => 1,
+                'personalIdentityCode' => '11111111111',
+                'privileges' => [],
+            ])->mergeRecursive($tolkevaravPayload)->toArray(),
+        ];
+
+        return static::createJwt($payload);
+    }
```

Comment:
**[non-blocker]** Attempted to use this in another branch. Found that the merging does not behave as you would expect.

```
collect([
    'userId' => 1,
    'personalIdentityCode' => 11111111111,
    'privileges' => []
])->mergeRecursive([
    'userId' => 'f2c97943-d036-4a82-baa6-d729d81fe47e',
    'personalIdentityCode' => "50608024740",
    'privileges' => ['ADD_ROLE']
])->toArray();
```

```
[
    "userId" => [
      1,
      "f2c97943-d036-4a82-baa6-d729d81fe47e",
    ],
    "personalIdentityCode" => [
      11111111111,
      "50608024740",
    ],
    "privileges" => [
      "ADD_ROLE",
    ],
  ]
```

Reply:
fixed